packages/swirl-components/src/components/swirl-image-grid-item/swirl-image-grid-item.tsx

```diff
@@ -90,6 +114,45 @@ export class SwirlImageGridItem {
     this.imageError.emit();
   };
 
+  private pauseGif = () => {
+    this.gifPaused = true;
+    const imageEl = this.img;
+    this.readImageFromCanvas(imageEl.src).then((image) => {
+      imageEl.src = image;
+      this.backgroundImg.attributeStyleMap.set(
+        "background-image",
+        `url(${image})`
+      );
+    });
+  };
+
+  private playGif = () => {
+    this.gifPaused = false;
+    const imageEl = this.img;
+    imageEl.src = this.src;
+    this.backgroundImg.attributeStyleMap.set(
+      "background-image",
+      `url(${this.src})`
+    );
+  };
+
+  readImageFromCanvas(imageSrc: string): Promise<string> {
+    return new Promise((resolve) => {
+      const image: HTMLImageElement = new Image();
+      image.src = imageSrc;
+      image.onload = (ev) => {
+        const el = ev.target as HTMLImageElement;
+        const canvas = document.createElement("canvas"),
+          ctx = canvas.getContext("2d");
+        canvas.width = el.width;
+        canvas.height = el.height;
+        ctx.drawImage(el, 0, 0, el.width, el.height);
+        resolve(canvas.toDataURL("image/jpeg"));
+      };
+      image.onerror = () => resolve("");
```

Comment:
Issue: This would cause the component to show no image, right? Probably better to reject() the promise here and handle the error in `pauseGif()`.

Reply:
it just did nothing. i changed it to log an error now anyways